corgi/collectors/app_interface.py

```diff
@@ -11,7 +11,9 @@
 
 class AppInterface:
     @classmethod
-    def fetch_service_metadata(cls) -> dict[ProductStream, list[dict]]:
+    def fetch_service_metadata(
+        cls, services: list[ProductStream]
```

Comment:
To make mypy pass, I think services should have type QuerySet[ProductStream], or Iterable[ProductStream] if you need to pass both lists and querysets:
```suggestion
        cls, services: QuerySet[ProductStream]
```

Reply:
I made it pass a list.